### ISSUE TITLE
How to $filter->where() callback not called when value is zero (0) or empty string

### ISSUE THREAD

**Issue by isometriq** [z-song/laravel-admin#1016]
I would expect the callback to be called when my select box filter is selected to any value. Yes a textbox  may be different, but even so I see the ->where() method for customizing the filtering queries ..so it should not decide if empty string or number and just execute the callback.

**Comment by isometriq** [comment]
By using a ->equal() method, the empty value is considered, so its really the ->where() method.

**Comment by yccphp** [comment]
Yes i encountered the same problem
I found that this code is causing

```
public function condition($inputs)
    {
        $value = array_get($inputs, static::getQueryHash($this->where));

        if (is_array($value)) {
            $value = array_filter($value);
        }

        if (is_null($value) || empty($value)) {
            return;
        }

        $this->input = $this->value = $value;

        return $this->buildCondition($this->where->bindTo($this));
    }

```

It does not need these judgments
```
if (is_null($value) || empty($value)) {
            return;
        }
```

@z-song  Can you solve this problem?

Thank you

**Comment by z-song** [comment]
Fixed in bf346cf
